src/pages/index.js

```diff
@@ -0,0 +1,305 @@
+const editFormElement = editModal.querySelector(".modal__form");
+const editModalCloseButton = editModal.querySelector(".modal__close-button");
+const editModalNameInput = editModal.querySelector("#profile-name-input");
+const editModalDescriptionInput = editModal.querySelector(
+  "#profile-description-input"
+);
+
+//Avatar
+const avatarModal = document.querySelector("#avatar-modal");
+const avatarModalForm = avatarModal.querySelector(".modal__form");
+const avatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarModalInput = avatarModal.querySelector("#profile-avatar-input");
+const avatarElement = document.querySelector(".profile__avatar");
+
+//Delete
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalForm = deleteModal.querySelector(".modal__form-delete");
+const deleteModalCloseButton = deleteModal.querySelector(
+  ".modal__close-button_delete"
+);
+const deleteModalCancelButton = deleteModal.querySelector(
+  ".modal__submit-button_type_cancel"
+);
+
+//Modal
+const previewModal = document.querySelector("#preview-modal");
+const previewModalImageEl = previewModal.querySelector(".modal__image");
+const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
+const previewModalCloseButton = previewModal.querySelector(
+  ".modal__close-button"
+);
+const editModalCloseIcon = document.querySelectorAll(".modal__close-img");
+
+//Cards
+const cardModal = document.querySelector("#add-card-modal");
+const cardForm = cardModal.querySelector(".modal__form");
+const cardSubmitButton = cardModal.querySelector(".modal__submit-button");
+const cardModalCloseButton = cardModal.querySelector(".modal__close-button");
+const cardNameInput = cardForm.querySelector("#add-card-name-input");
+const cardLinkInput = cardForm.querySelector("#add-card-link-input");
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+let selectedCard, selectedCardId;
+
+function handleDeleteCardButton(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function getCardElement(data) {
+  console.log("Card data:", data);
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  const isLiked = data.isLiked;
+  if (isLiked) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  cardLikeButton.addEventListener("click", () => {
+    cardLikeButton.classList.toggle("card__like-button_liked");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.